### PR TITLE
Use infrastructure attributes for storage selection

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: Shlomit Koyfman <shlomitk@il.ibm.com>
Fixes https://github.com/fybrik/fybrik/issues/1220
Fixes https://github.com/fybrik/fybrik/issues/1252

Changes in this PR:
- Change to FybrikStorageAccount: a single region/endpoint per object store. This allows to select storage accounts based on the region.
Example:
```
spec:
  id: theshire-object-store
  secretRef: credentials-theshire
  region: theshire
  endpoint: http://s3.eu.cloud-object-storage.appdomain.cloud
```
- Change to the restriction. Check for a value in a set of values, or inside a range for numeric values.
Example: {"property": "storage-cost", "range": {"max": 90}}
- Infrastructure attribute, its value, definition and connection to a validated object. For example:
```
{
            "attribute": "storage-cost",
            "description": "theshire object store",
            "value": "90",
            "type": "numeric",
            "units": "US Dollar",
            "instance": "account-theshire"
}
```

To do in the next PRs: 
- changes to documentation
- infrastructure object validation
- sample taxonomy for attribute values

### CODE REVIEW

File: charts/fybrik/files/taxonomy/attributes.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (simanadler) [inline charts/fybrik/files/taxonomy/attributes.json:1]: Perhaps change file name to infra-attributes.json for clarity?

File: charts/fybrik/files/taxonomy/adminrules.json [hunk 2]
```diff
@@ -75,24 +75,55 @@
...
-        "$ref": "#/definitions/StringList"
+      "required": [
+        "property"
+      ],
+      "properties": {
+        "property": {
+          "type": "string"
+        },
+        "range": {
+          "$ref": "#/definitions/RangeType"
```
Comment (simanadler) [inline charts/fybrik/files/taxonomy/adminrules.json:99]: Range should be optional, right?

**Review by simanadler — COMMENTED** [review]
Really minor comments

File: charts/fybrik/files/taxonomy/adminrules.json (hunk 2 above)
Comment (shlomitk1) [reply charts/fybrik/files/taxonomy/adminrules.json:99]: It is optional (does not appear in the 'required' section, line 91)

File: charts/fybrik/files/taxonomy/attributes.json (hunk 1 above)
Comment (shlomitk1) [reply charts/fybrik/files/taxonomy/attributes.json]: The name of the file is auto-generated from the package name, and there are some lint rules for package naming. Thus, the result will be infraattributes.json

**Review by simanadler — APPROVED** [review]

**Pull request merged**